### ISSUE TITLE
Unsure what object-literal-shorthand rule is reporting here

### ISSUE THREAD

**Issue by chrismbarr** [palantir/tslint#1505]
I just updated to TSLint 3.15.0 and the new rule `object-literal-shorthand` seems to complain about this code:
```ts
const fullFilePath = sourceFile.fileName;
const lastSlashIdx = fullFilePath.lastIndexOf("/");
const filePath = fullFilePath.substr(0, lastSlashIdx + 1);
const fileName = fullFilePath.substr(lastSlashIdx + 1, fullFilePath.length);
const fileNameParts = fileName.split(".");
return {
    fileName: fileName,
    fileNameParts: fileNameParts,
    filePath: filePath
};
```

For each line in the `return` statement it throws: `Expected property shorthand in object literal.`

I'm not really sure what "shorthand" means in this context... ~~and there [is no documentation for this rule right now](http://palantir.github.io/tslint/rules/).~~

---

It seems that moving things from the variable declaration directly into the object will fix some of these warnings, but I can only do so much. 

```ts
const fullFilePath = sourceFile.fileName;
const lastSlashIdx = fullFilePath.lastIndexOf("/");
const fileName = fullFilePath.substr(lastSlashIdx + 1, fullFilePath.length);
return {
    fileName: fileName,
    fileNameParts: fileName.split("."),
    filePath: fullFilePath.substr(0, lastSlashIdx + 1)
};
```

Doing that now only makes it throw one error for the first line of the `return` statement, but if I do what I think it's asking me to do here, it make my code more complex, which is not something I want to do.

---

This code satisfies the linter, but lines 3 & 5 are now duplicated :sob: 
```ts
const fullFilePath = sourceFile.fileName;
const lastSlashIdx = fullFilePath.lastIndexOf("/");
const fileName = fullFilePath.substr(lastSlashIdx + 1, fullFilePath.length);
return {
    fileName: fullFilePath.substr(lastSlashIdx + 1, fullFilePath.length),
    fileNameParts: fileName.split("."),
    filePath: fullFilePath.substr(0, lastSlashIdx + 1)
};
```

I'll have to turn the rule off for now.

**Comment by adidahiya** [comment]
it's similar to this eslint rule: http://eslint.org/docs/rules/object-shorthand
more docs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer

we just updated the docs a few min ago: http://palantir.github.io/tslint/rules/object-literal-shorthand/

feel free to send PRs to improve that rule's metadata :)

**Comment by chrismbarr** [comment]
So this is the intended behavior then?

**Comment by adidahiya** [comment]
yes. you would probably be best off satisfying the rule this way:

```ts
return {
    fileName,
    fileNameParts,
    filePath
};
```

**Comment by chrismbarr** [comment]
ok, interesting.  I think I understand now.,  I'll have to look into this more. Thanks & sorry to both with a non-bug!
